apps/rush-lib/src/data/RushConfiguration.ts

```diff
@@ -575,9 +581,31 @@ export default class RushConfiguration {
 
     this._rushLinkJsonFilename = path.join(this._commonTempFolder, 'rush-link.json');
 
-    this._npmToolVersion = rushConfigurationJson.npmVersion;
-    this._npmToolFilename = path.resolve(path.join(this._commonTempFolder,
-      'npm-local', 'node_modules', '.bin', 'npm'));
+    if (rushConfigurationJson.npmVersion) {
+      this._packageManager = 'npm';
+
+      this._committedShrinkwrapFilename = path.join(this._commonRushConfigFolder, RushConstants.npmShrinkwrapFilename);
+      this._tempShrinkwrapFilename = path.join(this._commonTempFolder, RushConstants.npmShrinkwrapFilename);
+
+      this._packageManagerToolVersion = rushConfigurationJson.npmVersion;
+      this._packageManagerToolFilename = path.resolve(path.join(this._commonTempFolder,
+        'npm-local', 'node_modules', '.bin', 'npm'));
+
+    } else if (rushConfigurationJson.pnpmVersion) {
```

Comment:
You should throw if they're both defined.

Reply:
The schema should enforce this